<?xml version="1.0"?>
<launch>
  <arg name="dataRoot" default="$(env HOME)" />
  <arg name="file_extension" default=".yaml" />
  <arg name="tabletop_segmentation_service" value="tabletop_segmentation" />

  <include file="$(find pr2_social_gaze)/launch/gaze.launch" />
  <include file="$(find pr2_moveit_config)/launch/move_group.launch" />
  <include file="$(find tabletop_object_detector)/launch/tabletop_segmentation.launch"/>

  <node name="pr2_pbd_interaction" pkg="pr2_pbd_interaction" type="interaction.py" output="screen">
    <param name="dataRoot" value="$(arg dataRoot)" />
    <param name="file_extension" value="$(arg file_extension)" />
    <param name="actionsRoot" value="$(arg dataRoot)/pbd_manipulation_saved/" />
    <param name="tabletop_segmentation_service" value="$(arg tabletop_segmentation_service)" />
  </node>

  <!-- This is the stuff required for web-based GUI -->
  <include file="$(find pr2_interactive_controller)/launch/web_launch.launch"/>
  <node pkg="interactive_marker_proxy" type="proxy" name="proxy-world-objects" output="screen" args="topic_ns:=/world_objects target_frame:=/odom_combined" />
  <node pkg="interactive_marker_proxy" type="proxy" name="proxy-saved-poses" output="screen" args="topic_ns:=/programmed_actions target_frame:=/odom_combined" />
</launch>
